src/components/TradeWidget/TokenRow.tsx

```diff
@@ -226,9 +239,10 @@ const TokenRow: React.FC<Props> = ({
           required
           ref={register({
             validate: {
-              positive: (value: string): true | string => validatePositive(value),
+              positive: (value: string): true | string => validatePositive(value, selectedToken.decimals),
```

Comment:
I can't be sure, but
As `validate` inside register is memoized (isn't it?) and `selectedToken` comes from props, which can change. If `decimals` change, `validate` would still have older decimals.

Reply:
That is why I mentioned in some other comment just remounting the whole thing on token change.

Reply:
You are right. This needs fixing.